### PR TITLE
Used Address model to store client address instead of address column.

### PR DESCRIPTION
**Pull request by deepti-kakade**
**Notion:**

https://www.notion.so/saeloun/6afc9500b2cc42af86be55f37894fb09?v=185bd45b9dc24dfcad9c57c07d32a7c0&p=2587d88361084386a003bfee8c1f5029&pm=s

**Description:**

After developing the updated designs of the client creation page, backend changes are required for address fields on the new client page.

The address field is a text field that needs to be updated to different fields for Address line 1, Address line 2, Country, State, City & Zipcode.

- We used Address association to save the client's address.

- Modified create/update client API.

- As well as list client API impacted, so made changes in that API as well.

- Migrating old addresses into the Address model. Created a migration for the migration.

**Impacted API Endpoint:**

- GET - http://localhost:3000/internal_api/v1/clients?time_frame=week
- POST - http://localhost:3000/internal_api/v1/clients
- PATCH -http://localhost:3000/internal_api/v1/clients/<:client_id>
-  Generate Invoice API

### CODE REVIEW

File: spec/factories/addresses.rb [hunk 1]
```diff
@@ -14,6 +14,10 @@
       association(:addressable, factory: :company)
     end
 
+    trait :with_client do
```
Comment (shalapatil) [inline spec/factories/addresses.rb:17]: this also needs to be fixed to use the created client

**Review by shalapatil — CHANGES_REQUESTED** [review]

File: app/models/company.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # Table name: companies
 #
 #  id              :bigint           not null, primary key
-#  address         :text             not null
+#  address         :text
```
[the remark this answers is not in the document]
Comment (deepti-kakade) [reply app/models/company.rb:6]: @shalapatil 
The address was not null by default, so we changed the default value to null to allow null values.

File: app/views/internal_api/v1/clients/_client.json.jbuilder [hunk 3]
```diff
@@ -1,3 +1,8 @@
 # frozen_string_literal: true
 
-json.extract! client, :id, :name, :email, :phone, :address, :logo
+json.client do
+  json.extract! client, :id, :name, :email, :phone, :logo
+  json.address do
```
[the remark this answers is not in the document]
Comment (deepti-kakade) [reply app/views/internal_api/v1/clients/_client.json.jbuilder:5]: Okay

File: db/seeds.rb [hunk 4]
```diff
@@ -46,8 +46,12 @@
 puts "Employment Created"
 
 microsoft_client = company.clients.create!(
-  name: "Microsoft", email: "support@example.com", phone: "+1 9999999991",
-  address: "California, USA")
+  name: "Microsoft",
+  email: "support@example.com",
+  phone: "+1 9999999991"
+  addresses_attributes: [{
+    address_line_1: "475 Clermont Ave", city: "Brooklyn", state: "NY", pin: "12238", country: "USA"
```
[the remark this answers is not in the document]
Comment (deepti-kakade) [reply db/seeds.rb:53]: @shalapatil I think we can't use attributes_for method here because it is available with factory and Rspec.

**Comment by supriya3105** [comment]
@prasanthchaduvula Same for this PR. It is connected to the previous PR. Please review and make the required changes yourself to get it merged as soon as possible.

**Comment by prasanthchaduvula** [comment]
@supriya3105 is it regarding the flaky system spec or regarding the PR review comments

**Comment by supriya3105** [comment]
> @supriya3105 is it regarding the flaky system spec or regarding the PR review comments

@prasanthchaduvula Review comments . If there are flaky specs that are failing , please fix them too. Anything that's required to get the PR merged.

**Comment by prasanthchaduvula** [comment]
Closing this PR as these changes already got merged https://github.com/saeloun/miru-web/pull/1240

**Pull request closed without merging**